x/meshsecurity/client/cli/gov_tx.go

```diff
@@ -0,0 +1,136 @@
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/address"
+	"github.com/cosmos/cosmos-sdk/version"
+	"github.com/cosmos/cosmos-sdk/x/gov/client/cli"
+	v1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/osmosis-labs/mesh-security-sdk/x/meshsecurity/types"
+)
+
+// DefaultGovAuthority is set to the gov module address.
+// Extension point for chains to overwrite the default
+var DefaultGovAuthority = sdk.AccAddress(address.Module("gov"))
+
+func SubmitProposalCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "submit-proposal",
+		Short:        "Submit a mesh security proposal.",
+		SilenceUsage: true,
+	}
+	cmd.AddCommand(
+		ProposalSetVirtualStakingMaxCapCmd(),
+	)
+	return cmd
+}
+
+func ProposalSetVirtualStakingMaxCapCmd() *cobra.Command {
+	bech32Prefix := sdk.GetConfig().GetBech32AccountAddrPrefix()
+	cmd := &cobra.Command{
+		Use:   "set-virtual-staking-max-cap [contract_addr_bech32] [max_cap] --title [text] --summary [text] --authority [address]",
+		Short: "Submit a set virtual staking max cap proposal",
+		Args:  cobra.ExactArgs(2),
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Submit a proposal to set virtual staking maximum cap limit to the given contract.
+
+Example:
+$ %s tx meshsecurity submit-proposal set-virtual-staking-max-cap %s1l94ptufswr6v7qntax4m7nvn3jgf6k4gn2rknq 100stake --title "a title" --summary "a summary" --authority %s1eppn90g4eg7guvyhz0ayph40c3e6uktygdd2nq
+`, version.AppName, bech32Prefix, bech32Prefix)),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, proposalTitle, summary, deposit, err := getProposalInfo(cmd)
+			if err != nil {
+				return err
+			}
+			authority, err := cmd.Flags().GetString(flagAuthority)
+			if err != nil {
+				return fmt.Errorf("authority: %s", err)
+			}
+
+			if len(authority) == 0 {
+				return errors.New("authority address is required")
+			}
+
+			src, err := parseSetVirtualStakingMaxCapArgs(args, authority)
+			if err != nil {
+				return err
+			}
+
+			proposalMsg, err := v1.NewMsgSubmitProposal([]sdk.Msg{&src}, deposit, clientCtx.GetFromAddress().String(), "", proposalTitle, summary)
+			if err != nil {
+				return err
+			}
+			if err = proposalMsg.ValidateBasic(); err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), proposalMsg)
+		},
+		SilenceUsage: true,
+	}
+
+	// proposal flags
+	addCommonProposalFlags(cmd)
+	return cmd
+}
+
+func parseSetVirtualStakingMaxCapArgs(args []string, authority string) (types.MsgSetVirtualStakingMaxCap, error) {
+	maxCap, err := sdk.ParseCoinNormalized(args[1])
+	if err != nil {
+		return types.MsgSetVirtualStakingMaxCap{}, errorsmod.Wrap(err, "max cap")
+	}
+
+	msg := types.MsgSetVirtualStakingMaxCap{
+		Authority: authority,
+		Contract:  args[0],
+		MaxCap:    maxCap,
+	}
+	return msg, nil
+}
+
+func addCommonProposalFlags(cmd *cobra.Command) {
```

Comment:
I have noticed that there is an empty `Metadata` field when I query the proposal. Is this something we should set here, too?